backend/cn/lib/compile.ml

```diff
@@ -24,11 +24,6 @@ module STermMap = Map.Make (struct
 module StringMap = Map.Make (String)
 module StringSet = Set.Make (String)
 
-type function_sig =
-  { args : (Sym.t * BaseTypes.t) list;
-    return_bty : BaseTypes.t
-  }
-
```

Comment:
It would be cleaner (I think) to just keep this type and convert the Builtins to this rather than converting giving dummy definitions to other functions, though if this is not the case I would like to know.

Reply:
In `core_to_mucore.ml`, logical functions from the frontend are converted to `lfuns` via `translate_cn_function` and then put in the terribly named and should be renamed `mu_logical_predicates`. Those are then read in `check.ml` by being passed to `record_and_check_logical_functions`. From there, they are added (first without definitions, for mutual recursion), type-checked, and then have there definitions added in after type-checking. If the definition is needed, then `get_logical_function_def` is called on the `Global.t` typing environment.

`register_cn_function` should only add the signatures (so this can handle builtin and user-defined functions at the same time) but `translate_cn_function` converts the definitions for user-defined functions, you just prepend the builtins to `lfuns` and the rest will work out fine.